common/src/main/java/io/apicurio/registry/utils/Maybe.java

```diff
@@ -0,0 +1,97 @@
+package io.apicurio.registry.utils;
+
+import java.util.NoSuchElementException;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+/**
+ * Proxyable alternative to java.util.Optional.
+ * Optional cannot be used with CDI because it is final.
+ *
+ * @author Jakub Senko <jsenko@redhat.com>
+ */
+public class Maybe<T> {
```

Comment:
While I understand the need for this class, I don't really like the name since Maybe has some implications in functional programming which this class do not have. That said, I don't have a good suggestion other than perhaps `Option`.

Reply:
I understand, but I don't like `Option` much, because it's an overloaded term (configuration option) and I like the `Optional` name they've decided to use in Java. `Maybe` is used in e.g. Haskell, but for the same purpose and I don't think someone would be confused and expect it behave exactly as in Haskell.

Reply:
I suggest renaming this to `OptionalBean`, `ConditionalBean`, `Conditional` , `Reference` , `OptionalReference` ,...

Reply:
Yeah, I don't like `Option` too much either. I was making that suggestion since it's the name used in Scala.